rubber_duck.py

```diff
@@ -132,25 +137,92 @@ async def have_conversation(self, thread_id: int, engine: str, prompt: str, init
                         f'\n{ex}\n'
                         '\n'.join(tb.format_exception(ex))
                     )
+
+                    # For server-side errors
+                    if isinstance(ex,
+                                  (openai.APITimeoutError, openai.InternalServerError, openai.UnprocessableEntityError)):
+                        await self._edit_message(thread_id, self._error_message_id,
+                                                 'I\'m having trouble connecting to the OpenAI servers, '
+                                                 'please open up a separate conversation and try again')
+                    # For client-side errors
+                    elif isinstance(ex, (openai.APIConnectionError, openai.BadRequestError,
+                                         openai.AuthenticationError, openai.ConflictError, openai.ConflictError, openai.NotFoundError,
+                                         openai.RateLimitError)):
+                        user_ids_to_mention = [911012305880358952, 933123843038535741, 1014286006595358791,
+                                               353454081265762315, 941080292557471764] #Dr.Bean, MaKenna, Chase, YoungWoo, Molly's ID's
+                        mentions = ' '.join([f'<@{user_id}>' for user_id in user_ids_to_mention])
+                        openai_web_mention = "Visit https://platform.openai.com/docs/guides/error-codes/api-errors " \
+                                             "for more details on how to resolve this error"
+                        await self._edit_message(thread_id, self._error_message_id,
+                                                 'I\'m having trouble processing your request, '
+                                                 'I have notified your professor to look into the problem!')
+                        if isinstance(ex, openai.APIConnectionError):
+                            client_error_message = "*** APIConnectionError ***"
+                        elif isinstance(ex, openai.BadRequestError):
+                            client_error_message = "*** BadRequestError ***"
+                        elif isinstance(ex, openai.AuthenticationError):
+                            client_error_message = "*** AuthenticationError ***"
+                        elif isinstance(ex, openai.ConflictError):
+                            client_error_message = "*** ConflictError ***"
+                        elif isinstance(ex, openai.NotFoundError):
+                            client_error_message = "*** NotFoundError ***"
+                        elif isinstance(ex, openai.PermissionDeniedError):
+                            client_error_message = "*** PermissionDeniedError ***"
+                        elif isinstance(ex, openai.RateLimitError):
+                            client_error_message = "*** RateLimitError ***"
```

Comment:
Just use the `error_message` from above. It will have this information in it already.

But for the record, you could:
```python
client_error_message = f'*** {ex.__name__} ***'
```